torch/_dynamo/variables/builder.py

```diff
@@ -1063,7 +1063,8 @@ def wrap_to_fake_tensor_and_record(
     if type(e) in (torch.Tensor, torch.nn.Parameter) or (
         ignore_subclass and isinstance(e, torch.Tensor)
     ):
-        static_shapes, reason = tensor_always_has_static_shape(e, source, is_tensor)
+        assert source is not None
```

Comment:
Assert this at the very beginning of the function and change the mypy annotation

Reply:
I am not sure this assert is valid for non tensor cases?